common/test_raw_ostream.h

```diff
@@ -0,0 +1,49 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_COMMON_TEST_RAW_OSTREAM_H_
+#define CARBON_COMMON_TEST_RAW_OSTREAM_H_
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "common/ostream.h"
+
+namespace Carbon::Testing {
+
+/// A raw_ostream that makes it easy to repeatedly check streamed output.
+class TestRawOstream : public llvm::raw_ostream {
```

Comment:
If you're trying to share this, what would you think of:

```
class TestRawOstream : public llvm::raw_string_ostream {
 public:
  explicit TestRawOstream() : llvm::raw_string_ostream(buffer_);

 private:
  std::string buffer_;
};
```

But also, how much is this class adding over raw_string_ostream?

Reply:
Good suggestion, done. Also done the consequent simplifications.

The primary utility of this over `raw_string_ostream` is specifically `TakeStr()` as it makes it easy to do a series of operations interleaved with `EXPECT_THAT(s.TakeStr(), ...)`, where each expectation both checks the previous operation and resets the buffer to a clean slate for the next one.

The destructor is also nice to make sure that a trailing expectation isn't missed in a test.